### ISSUE TITLE
[css-masking-1] Clarification on mask-type presentation attribute

### ISSUE THREAD

**Issue by sandgraham** [w3c/fxtf-drafts#480]
I was hoping to clarify if `mask-type` should be assignable as a DOM attribute.

From the [spec](https://drafts.fxtf.org/css-masking/#the-mask-type):

> The [mask-type](https://drafts.fxtf.org/css-masking/#propdef-mask-type) property is a [presentation attribute](https://www.w3.org/TR/2011/REC-SVG11-20110816/intro.html#TermPresentationAttribute) for SVG elements.

The definition of a **presentation attribute** from the [SVG 1.1 Spec](https://www.w3.org/TR/2011/REC-SVG11-20110816/intro.html#TermPresentationAttribute):
> An XML attribute on an SVG element which specifies a value for a given [property](https://www.w3.org/TR/2011/REC-SVG11-20110816/intro.html#TermProperty) for that element.

From the styling section of the same spec, [6.4 Specifying properties using the presentation attributes](https://www.w3.org/TR/2011/REC-SVG11-20110816/styling.html#UsingPresentationAttributes), this is perhaps also relevant:
> For each styling property defined in this specification (see [Property Index](https://www.w3.org/TR/2011/REC-SVG11-20110816/propidx.html)), there is a corresponding XML attribute (the presentation attribute) with the same name that is available on all relevant SVG elements.

This leads me to believe that all presentation attributes should be available as an attribute on applicable DOM elements- meaning the following is valid:

```html
<mask id="mask" mask-type="alpha">
  <path d="..." />
</mask>
``` 

However, the examples shown in the draft masking spec seem to make a point of using inline style attributes:

```html
<mask id="mask" style="mask-type: alpha;">
  <path d="..." />
</mask>
```

I noticed that `mask-type` is not listed under the [presentation attributes](https://drafts.fxtf.org/css-masking/#the-mask-type) for the `mask` element (which seems contradictory), nor is it included in the [`SVGMaskElement` interface](https://drafts.fxtf.org/css-masking/#InterfaceSVGMaskElement).

This leads me to believe that `mask-type` might not be intended as a DOM attribute but I'd appreciate any guidance available on this topic.

Thanks!

**Comment by fsoder** [comment]
...snip...
> This leads me to believe that all presentation attributes should be available as an attribute on applicable DOM elements- meaning the following is valid:
> 
> ```
> <mask id="mask" mask-type="alpha">
>   <path d="..." />
> </mask>
> ```

Correct.

> Looking at this [codepen example](https://codepen.io/sandgraham/pen/BaVwwza?editors=1100), it seems Chrome, Firefox, and Safari accept the attribute.
> 
> However, the examples shown in the draft masking spec seem to make a point of using inline style attributes:
> 
> ```
> <mask id="mask" style="mask-type: alpha;">
>   <path d="..." />
> </mask>
> ```
> 
> And indeed, MDN [does the same](https://developer.mozilla.org/en-US/docs/Web/CSS/mask-type).

What mechanism examples use does not really carry any relevance. I.e they are not normative.

> I also noticed that `mask-type` is not listed under the [presentation attributes](https://drafts.fxtf.org/css-masking/#the-mask-type) for the `mask` element (which _seems_ contradictory),

This looks like an oversight/bug.

> nor is it included in the [`SVGMaskElement` interface](https://drafts.fxtf.org/css-masking/#InterfaceSVGMaskElement).

It's not uncommon for a presentation attribute to not have a corresponding IDL attribute (most on SVG elements don't).

> This leads me to believe that `mask-type` might not be intended as a DOM attribute. I'd appreciate any guidance available on this topic.

The term "DOM attribute" may be a bit ambiguous, so I'd suggest using the terminology from [1]. Using that terminology I'd say that there is a _content attribute_ `mask-type`, but it has no corresponding _IDL attribute_ (which would likely have been named `maskType` or `masktype` if it existed). Does that make sense?

[1] https://html.spec.whatwg.org/multipage/infrastructure.html#terminology

**Comment by ljharb** [comment]
@fsoder so that means HTML like `<mask mask-type="…">` would work, but the resulting DOM element wouldn't have any JS property that corresponded? (although perhaps `el.getAttribute('mask-type')` would work)

**Comment by fsoder** [comment]
Yes (assuming the ` <mask ...>` is defined such that it is in the SVG namespace of course). Reading using  `getAttribute()` would work, yes.

**Comment by ljharb** [comment]
Thanks!

for some feedback: from a reader's perspective, this style of specification (where HTML attributes and/or DOM element properties are not explicitly listed, but must be inferred) is very confusing and hard to understand (and thus, likely easier to implement incorrectly).
